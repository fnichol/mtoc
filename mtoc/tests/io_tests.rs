// Copyright 2019 Fletcher Nichol and/or applicable contributors.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license (see <LICENSE-MIT or
// http://opensource.org/licenses/MIT>, at your option. This file may not be copied, modified, or
// distributed except according to those terms.

mod helper;

use assert_cmd::assert::Assert;
use assert_cmd::prelude::*;
use helper::*;
use insta::assert_snapshot_matches;
use std::str;

macro_rules! test {
    (
        $name:ident, $fixture_file:expr
    ) => {
        #[test]
        fn $name() {
            let cmd = mtoc().arg($fixture_file).assert().success().stderr("");

            assert_snapshot_matches!(format!("{}.stdout", $fixture_file), stdout(&cmd));
        }
    };
}

fn stdout(cmd: &Assert) -> &str {
    str::from_utf8(&cmd.get_output().stdout).unwrap()
}

test!(simple_new, "simple-new.md");
test!(simple_old, "simple-old.md");
test!(simple_empty, "simple-empty.md");
test!(complex, "complex.md");
