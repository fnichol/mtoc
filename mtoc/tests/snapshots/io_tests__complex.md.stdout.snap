---
created: "2019-06-17T18:49:02.638268987Z"
creator: insta@0.8.1
source: mtoc/tests/io_tests.rs
expression: stdout(&cmd)
---
# Title

<!-- toc -->

- [jonschlinkert](#jonschlinkert)
  * [Foo](#foo)
  * [Foo](#-foo)
  * [Foo](#-foo-)
  * [Foo](#foo-)
  * [Some Article](#some-article)
  * [存在，【中文】；《标点》、符号！的标题？](#存在中文标点符号的标题)
  * [かんじ](#かんじ)
  * [中文](#中文)
  * [한자](#한자)
- [sebdah](#sebdah)
  * [Header %something%](#header-something)
  * [Header #something](#header-something-1)
- [naokazuterada](#naokazuterada)
  * [&lt;element1>](#ltelement1)
  * [(a static function) `greet([name])` (original, right?)](#a-static-function-greetname-original-right)
  * [_x_ test 2](#_x_-test-2)
- [jch](#jch)
  * [日本語](#日本語)
  * [Русский](#русский)
- [fnichol](#fnichol)
  * [Okay Åô Then](#okay-åô-then)
  * [Frachtaufträge](#frachtaufträge)
  * [C](#c)
  * [On [Go to Top](#top)](#on-go-to-toptop)

<!-- tocstop -->

## jonschlinkert

### <test>Foo

### <test> Foo

### <div> Foo </div>

### Foo <test>

### Some Article

### 存在，【中文】；《标点》、符号！的标题？

### かんじ

### 中文

### 한자

## sebdah

### Header %something%

### Header #something

## naokazuterada

### &lt;element1>

### (a static function) `greet([name])` (original, right?)

### _x_ test 2

## jch

### 日本語

### Русский

## fnichol

### Okay Åô Then

### Frachtaufträge

### C

### On <sub><sub>[Go to Top](#top)</sub></sub>

